test/Splunk.OpenTelemetry.AutoInstrumentation.Tests/LoggerTests.cs

```diff
@@ -0,0 +1,37 @@
+// <copyright file="LoggerTests.cs" company="Splunk Inc.">
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Splunk.OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace Splunk.OpenTelemetry.AutoInstrumentation.Tests;
+
+public class LoggerTests
+{
+    [Fact]
+    public void ConstructorDoesNotThrowExceptionWhenReflectionFails()
+    {
+        var exception = Record.Exception(() => new Logger());
+        exception.Should().Be(null);
+    }
+
+    [Fact]
+    public void ImplementationDoesNotThrowExceptionWhenReflectionFails()
+    {
+        ILogger logger = new Logger();
+        var exception = Record.Exception(() => logger.Warning("message"));
```

Comment:
I would like to have verify the reflection code. 
This test would pass if it the Logger would be not able to initialize.

How will we detect a problem if `OpenTelemetry.AutoInstrumentation.Logging.OtelLogging` API changes?

https://github.com/signalfx/splunk-otel-dotnet/blob/1c0f981b93de30307a853555396c0b2e8aefb48f/src/Splunk.OpenTelemetry.AutoInstrumentation/Logging/Logger.cs#L45

Reply:
Logger is not initialized in this case as dll with unit tests does not have access to OtelAuto dll. The exception is caught and instrumentation still works correctly.

Reply:
> The exception is caught and instrumentation still works correctly.

If you want to test only this then you could create

```csharp
class LoggerSpy : ILogger {
   ILogger logger;
   bool called;

   LoggerSpy(ILogger logger) { this.logger = logger; }

    void Warning(string message)
    {
        this.logger.Warning(message);
        this.called = true;
    }
}
```

and use `new LoggerSpy(new Logger())` instead of `new Mock<ILogger>()` it in `ServiceNameWarningTests`

Reply:
This is separate class that `ServiceNameWarningTests` is independent of, I didn't want to merge it together.